packages/medusa/src/services/discount-condition.ts

```diff
@@ -0,0 +1,238 @@
+import { MedusaError } from "medusa-core-utils"
+import { BaseService } from "medusa-interfaces"
+import { EntityManager } from "typeorm"
+import { EventBusService } from "."
+import { Discount, DiscountCondition, DiscountConditionType } from "../models"
+import { DiscountRepository } from "../repositories/discount"
+import { DiscountConditionRepository } from "../repositories/discount-condition"
+import { DiscountRuleRepository } from "../repositories/discount-rule"
+import { UpsertDiscountConditionInput } from "../types/discount"
+import { PostgresError } from "../utils/exception-formatter"
+
+/**
+ * Provides layer to manipulate discount conditions.
+ * @implements {BaseService}
+ */
+class DiscountConditionService extends BaseService {
+  protected readonly manager_: EntityManager
+  protected readonly discountRuleRepository_: typeof DiscountRuleRepository
+  protected readonly discountRepository_: typeof DiscountRepository
+  protected readonly discountConditionRepository_: typeof DiscountConditionRepository
+  protected readonly eventBus_: EventBusService
+
+  constructor({
+    manager,
+    discountRuleRepository,
+    discountRepository,
+    discountConditionRepository,
+    eventBusService,
+  }) {
+    super()
+
+    /** @private @const {EntityManager} */
```

Comment:
we should remove those comment that does not correspond to there visibility anymore